pandas/tests/resample/test_resample_api.py

```diff
@@ -771,3 +771,58 @@ def test_end_and_end_day_origin(
     )
 
     tm.assert_series_equal(res, expected)
+
+
+@pytest.mark.parametrize("numeric_only", [True, False])
+def test_downsample_method(numeric_only):
+    # test if `numeric_only` behave as expected for Resampler downsample methods.
+
+    index = date_range("2018-01-01", periods=2, freq="D")
+    expected_index = date_range("2018-12-31", periods=1, freq="Y")
+    df = DataFrame({"cat": ["cat_1", "cat_2"], "num": [5, 20]}, index=index)
+    resampled = df.resample("Y")
+
+    # test Resampler.sum
+    result = resampled.sum(numeric_only=numeric_only)
+    if numeric_only:
+        expected = DataFrame({"num": [25]}, index=expected_index)
+    else:
+        expected = DataFrame({"cat": ["cat_1cat_2"], "num": [25]}, index=expected_index)
+    tm.assert_frame_equal(result, expected)
```

Comment:
Test looks good, but looks to me like you can parametrize it; something like

```
@pytest.mark.parametrize(
    "how, expected_data",
    [
        ("sum", {"cat": ["cat_1cat_2"], "num": [25]}),
        ("prod", {"num": [100]}),
        ...
    ]
)
```

Then in the test:

```
if numeric_only:
    expected = expected[["num"]]
```

If I'm missing something and it'd be difficult to parametrize, okay as-is.

Reply:
Thanks @rhshadrach. I will adopt this suggestion to my code.